specutils/analysis/width.py

```diff
@@ -0,0 +1,179 @@
+import numpy as np
+from astropy.stats.funcs import gaussian_sigma_to_fwhm
+from ..spectra import SpectralRegion
+from ..manipulation import extract_region
+from . import centroid
+
+
+__all__ = ['gaussian_sigma_width', 'gaussian_fwhm', 'fwhm']
+
+
+def _computation_wrapper(func, spectrum, region):
+
+    # No region, therefore whole spectrum.
+    if region is None:
+        return func(spectrum)
+
+    # Single region
+    elif isinstance(region, SpectralRegion):
+        return func(spectrum, region=region)
+
+    # List of regions
+    elif isinstance(region, list):
+        return [func(spectrum, region=reg) for reg in region]
+
+
+def gaussian_sigma_width(spectrum, region=None):
+    """
+    Estimate the width of the spectrum using a second-moment analysis, scaled
```

Comment:
Maybe split this into a short initial sentence? E.g. "Estimate the width of the spectrum (or region) using a second-moment analysis.  This estimate is scaled to ..."

This is useful mainly because the sphinx docs will use the first sentence to show a summary of the function.

(and similar for the FWHM)